sylvia-derive/src/contract/communication/reply.rs

```diff
@@ -195,29 +187,66 @@ struct ReplyData<'a> {
     /// Unique identifier for the reply.
     pub reply_id: Ident,
     /// Unique name of the handler from which the [reply_id](ReplyData::reply_id) was constructed.
-    pub handler_name: &'a Ident,
-    /// Handler methods for the reply id.
-    pub handlers: Vec<&'a MsgVariant<'a>>,
+    pub handler_id: &'a Ident,
+    /// Methods handling the reply id for the associated reply on.
+    pub handlers: Vec<(&'a Ident, ReplyOn)>,
+    /// Payload parameters associated with the handlers.
+    pub payload: Vec<&'a MsgField<'a>>,
 }
 
 impl<'a> ReplyData<'a> {
-    pub fn new(reply_id: Ident, variant: &'a MsgVariant<'a>, handler_name: &'a Ident) -> Self {
+    pub fn new(reply_id: Ident, variant: &'a MsgVariant<'a>, handler_id: &'a Ident) -> Self {
+        // Skip the first field reserved for the `data`.
+        let payload = variant.fields().iter().skip(1).collect::<Vec<_>>();
+        let method_name = variant.function_name();
+        let reply_on = variant.msg_attr().reply_on();
+
         Self {
             reply_id,
-            handler_name,
-            handlers: vec![variant],
+            handler_id,
+            handlers: vec![(method_name, reply_on)],
+            payload,
         }
     }
 
+    /// Adds second handler to the reply data provdided their payload signature match.
+    pub fn add_second_handler(&mut self, new_handler: &'a MsgVariant<'a>) {
+        let (current_method_name, _) = match self.handlers.first() {
+            Some(handler) => handler,
+            _ => return,
+        };
+
+        if self.payload.len() != new_handler.fields().len() - 1 {
+            emit_error!(current_method_name.span(), "Mismatched lenght of method parameters.";
```

Comment:
typo: `length`. Better message would be `Methods arity mismatch.`

Reply:
I changed it to "number". I'm not sure how well known the "arity" word is and figured it would be easier to understand.